#! /bin/sh -e

# All lines beginning with `# DPATCH:' are a description of the patch.
# DP: don't choke on notes while reinserting REG_EH_REGION notes

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        #cd ${dir}gcc && autoconf2.59
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

---

 gcc/recog.c |    1 +
 1 file changed, 1 insertion(+)

Index: gcc-4.1/gcc/recog.c
===================================================================
--- gcc-4.1.orig/gcc/recog.c
+++ gcc-4.1/gcc/recog.c
@@ -3215,6 +3215,7 @@ peephole2_optimize (FILE *dump_file ATTR
 		      for (x = try ; x != before_try ; x = PREV_INSN (x))
 			if (CALL_P (x)
 			    || (flag_non_call_exceptions
+				&& INSN_P (x)
 				&& may_trap_p (PATTERN (x))
 				&& !find_reg_note (x, REG_EH_REGION, NULL)))
 			  {
